src/api/index.js

```diff
@@ -21,6 +21,36 @@ const METHOD = {
   POST: 'POST',
 };
 
+export const v4 = {
+  root: 'https://api.github.com/graphql',
+  call: async parameters => {
+    const response = await fetch(v4.root, parameters);
+    const json = JSON.parse(response._bodyText);
+
+    if (response.status >= 200 && response.status < 300) {
+      return json.data;
+    }
```

Comment:
Could we test for failed response.statuses instead before doing the JSON.parse() ?

Reply:
The GraphQL API returns JSON on error as well as success. I do the JSON parsing ahead of this line so I can get the message out of the body as to why the query failed, to put in the `Error`.

Reply:
Didn't know that, thanks for clarifying it, it's perfect that way then.

Reply:
I might be able to do `await response.json()` though instead of parsing the body myself, though. 🤔

Reply:
And it makes more sense along with Promise.reject() 👌